### PR TITLE
Rename functions like `genocide!`

### LINKED ISSUE #764: Renaming the `genocide!` function
Hello, 

I propose changing the name of the function `genocide!`.
This is because the word genocide is linked to heinous events in history and in our current times.
Therefore, to not normalise the word's meaning and the human suffering it signifies, I propose changing the name to something like `remove_all!`.

Similarly, the function `kill_agent!`could be renamed `remove_agent!`.

### PR DESCRIPTION
**Pull request by MA-Ramirez**
Closes #764  

genocide! --> remove_all!
kill_agent! --> remove_agent!
Unkillable --> Unremovable

### CODE REVIEW

**Comment by jacobusmmsmit** [comment]
Thank you for raising this, it has a strong +1 from me. Firstly for the obvious reason of how (euphemistically) _unnecessary_ the current names are, but secondly the current wording goes against the naming convention that functions that do the opposite things should have opposite names. Hence, I believe that `remove` is the correct opposite of `add` and not `kill`.

File: examples/optim.jl [hunk 1]
```diff
@@ -135,7 +135,7 @@
 # Unfortunately we've not given the optimiser information we probably needed to.
 # Notice that the death rate is 96%, with reinfection quite low.
 # When all the infected individuals die, infection doesn't transmit - the optimiser has
-# managed to reduce the infection rate by killing the infected.
+# managed to reduce the infection rate by removing the infected.
```
Comment (jacobusmmsmit) [inline examples/optim.jl:138]:
```suggestion
# managed to reduce the infection rate by "removing" the infected.
```
Makes it sound more ominous and closer to the original joke.

File: src/spaces/graph.jl [hunk 2]
```diff
@@ -130,15 +130,15 @@ export rem_node!, add_node!, rem_vertex!, add_vertex!, add_edge!, rem_edge!
 
 """
      rem_node!(model::ABM{<: GraphSpace}, n::Int)
-Remove node (i.e. position) `n` from the model's graph. All agents in that node are killed.
+Remove node (i.e. position) `n` from the model's graph. All agents in that node are removed.
```
Comment (jacobusmmsmit) [inline src/spaces/graph.jl:133]:
```suggestion
Remove node (i.e. position) `n` from the model's graph. All agents in that node are removed from the model.
```

File: src/spaces/graph.jl [hunk 3]
```diff
@@ -161,7 +161,7 @@ end
 
 """
     rem_vertex!(model::ABM{<:GraphSpace}, n::Int)
-Remove node (i.e. position) `n` from the model's graph. All agents in that node are killed.
+Remove node (i.e. position) `n` from the model's graph. All agents in that node are removed.
```
Comment (jacobusmmsmit) [inline src/spaces/graph.jl:164]:
```suggestion
Remove node (i.e. position) `n` from the model's graph. All agents in that node are removed from the model.
```

**Review by jacobusmmsmit — CHANGES_REQUESTED** [review]
There are a few mentions of kill left in the docs.

**Comment by MA-Ramirez** [comment]
@jacobusmmsmit thank you for the additional modifications. I thought all the files in docs were automatically generated by the scripts, so I didn't check them. 
Is there something I can do to allow the push to my branch? Or should I make the changes in the branch myself?

**Comment by jacobusmmsmit** [comment]
Honestly, I don't have the definitive answer to either of your comments.
Wrt docs, your comment makes sense and if that's the case then I guess it LGTM.
Wrt git, I don't think you need to do anything and the problem seems to be on my end. Maybe someone else can suggest why I can't push to her branch?

**Review by Datseris — CHANGES_REQUESTED** [review]
fantastic, thank a lot. However, if I were to merge this as it is right now, it would break every single Agents.jl code out there because we are renaming functions that thousands of users use. You need to go to the src/deprecations.jl file and add deprecations for all changed API names. For example,

```
@deprecate UnkillableABM UnremovableABM
```
and so on and so forth. Also, please add a changelog entry for every API name that was changed.

Changes in the wording of docstrings or documentation pages do not need to be mentioned. Only changes in API names, i.e., names explicitly participating in `export` statements.

**Review by jacobusmmsmit — APPROVED** [review]
LGTM

**Comment by jacobusmmsmit** [comment]
Wait hang on, shouldn't these changes go in v5.9 not v5.8 as it's already been released?

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,9 @@
 - A memory bug about offsets calculation has been solved; besides, the `calculate_offsets` function has been sped-up by a significant amount.
 
 # v5.8
+- New function `remove_all!` replaces `genocide!` (now deprecated).
```
Comment (jacobusmmsmit) [inline CHANGELOG.md:6]: Put these new changes in `main`

**Review by jacobusmmsmit — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (jacobusmmsmit) [reply CHANGELOG.md:6]: I hope my "request changes" doesn't seem too much like I'm bossing you around @MA-Ramirez! I don't like telling people what to do when I'm not super confident that it's the right thing, but I'm pretty sure `main` is the right place for these changes :')

**Review by Datseris — APPROVED** [review]

**Comment by Datseris** [comment]
image

**Comment by xlejx-rodsxn** [comment]
Ah, weird. how did I end up in that version? anyway, thanks for taking care of this issue and the time invested in explaining.

**Pull request merged**